### PR TITLE
test: strengthen tests for `bessely0` function

### PR DESCRIPTION
**Pull request by parth121101**
As suggested by @Pranavchiku I have added more tests for bessel_y0 function for rigorous testing.

### CODE REVIEW

File: integration_tests/intrinsics_197.f90 [hunk 1]
```diff
@@ -61,5 +61,92 @@ program intrinsic_197
     print *, bessel_j0(0.1727272919_sp)
     if(bessel_j0(0.1727272919_sp) - (0.992555201) > 1e-12) error stop
 
+    print *, bessel_y0(3.229995504589554d-153)
+    if(bessel_y0(3.229995504589554d-153) - (-223.60567312692984) > 1e-12) error stop
+
+    print *, bessel_y0(1.9201212555874555d-153)
+    if(bessel_y0(1.9201212555874555d-153) - (-223.93677423698244) > 1e-12) error stop
```
Comment (Pranavchiku) [inline integration_tests/intrinsics_197.f90:68]: let's not do it this way, create two arrays `x` and `expected`, populate it and then test those values. You may refer integration test for `besselj0`

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Requested changes above.

**Comment by parth121101** [comment]
> Requested changes above.

Done @Pranavchiku, thanks for letting me know a better way to manage this tests

**Comment by Pranavchiku** [comment]
Would you mind adding test for large positive values?

**Comment by parth121101** [comment]
> Would you mind adding test for large positive values?

sure

File: integration_tests/intrinsics_197.f90 [hunk 2]
```diff
@@ -1,65 +1,59 @@
...
+          29.82407037185126_dp, 35.2459016393_dp, 46.2634946_dp, 24.7740903638_dp, &
+          15.8978408636372_dp, 8.801468212714914_dp, 8.798080767692923_dp, 8.791287884846061575_dp, &
+          0.6635324560331525_dp, 0.6633115808685429_dp, 0.6630906573122884_dp, 0.6628696853957159_dp, &
+          0.5042349934053336_dp, 0.5041796815747646_dp, 0.5041242821300232_dp, 0.50401322049655_dp, &
+          1.3994502099160337d-303, 1.199530179928029d-303, 9.99610149940024d-304, 5.9977008996401444d-304, &
+          499.19032387045183_dp, 499.2802878848461_dp, 499.3702518992403_dp, 499.5501799280288_dp]
+
+    expected = [ 7.9836428649154907E-155_dp,-7.0751499885966513E-155_dp, 4.4632457212022053E-155_dp, -7.7717941781117305E-156_dp, & 
+        -2.1476418943648812E-002_dp,  0.45752410314844527_dp, -3.6134118144921369E-003_dp, -1.1810118244863275_dp, &
+          5.7258225070763252E-002_dp, -1.9980991452871046E-002_dp,  0.27136794388300378_dp,  0.26703177612919299_dp, &
```
Comment (Pranavchiku) [inline integration_tests/intrinsics_197.f90:23]: same comment here

**Review by Pranavchiku — APPROVED** [review]
Minor comments, do that and this is good to merge.

File: integration_tests/intrinsics_197.f90 [hunk 3]
```diff
@@ -1,65 +1,59 @@
...
-
-
+    real(dp) :: x(40)
+    real(dp) :: expected(40)
+    integer :: i
+
+    real :: y(28)
+    real :: expected_y(28)
+
+    x = [9.98600559790084e307_dp,9.98800479820072e307_dp,9.9900039985006e307_dp,9.99200319880048e307_dp, &
```
[the remark this answers is not in the document]
Comment (certik) [reply integration_tests/intrinsics_197.f90:10]: It should be `123.456e9_dp`, not `d`.

**Review by certik — APPROVED** [review]

**Pull request merged**